### PR TITLE
Fix for 2483 Author and description null checks & tests

### PR DESCRIPTION
**Pull request by robertmuehsig**
Fix for #2483

I enhanced the PackageValidation for the "null-cases" for the description and author element. The NuGet package author itself is never (or at least currently) null, but maybe filled with an empty element. 

No <description> element set (or an empty description element will trigger the same behavior):

image

No <authors> or <owners> element set:

image

Without those checks some other processes might fail (e.g. the lucene job needs the description and might fail if it is null).

Currently you could upload something without the author tag, but when you try to update it you will get a validation error:

image

This PR should fix the null handling and discard any uploads without a description or an empty author.

It seems that a empty <description></description> is also catched, because somewhere in the pipeline the description will be set to null instead of string.empty in this case, so even if you submit an empty description, the code will discard the upload.

### CODE REVIEW

File: src/NuGetGallery/Services/PackageService.cs [hunk 1]
```diff
@@ -579,15 +579,23 @@ private static void ValidateNuGetPackageMetadata(PackageMetadata packageMetadata
                     throw new EntityException(Strings.NuGetPackageReleaseVersionContainsOnlyNumerics, "Version");
                 }
             }
-            if (packageMetadata.Authors != null && packageMetadata.Authors.Flatten().Length > 4000)
+            if (packageMetadata.Authors == null || packageMetadata.Authors.Flatten().Length == 0)
```
Comment (skofman1) [inline src/NuGetGallery/Services/PackageService.cs]: We currently allow pushing packages with empty authors. This shouldn't change

**Comment by robertmuehsig** [comment]
Regarding empty authors: Uploading without authors is ok, but changing the package via the edit form would still require a author? Or is this field no longer required?

**Comment by robertmuehsig** [comment]
I removed the author validation. So if you upload a package without any owner or author information it will pass, but if you ever try to update it via the UI the field is marked as "required".

**Comment by robertmuehsig** [comment]
@skofman1 is there anything still open from your review or do you think it is ok?

**Comment by skofman1** [comment]
Looks great! Thanks @robertmuehsig !

**Pull request merged**